### PR TITLE
Add first name, last name, city, state, and zip to registration

### PR DESCRIPTION
**Pull request by rwwiv**
## Changes
* Add city, state, zip_code to UserProfile model
* Add fields to registration form
* Create user profile during registration
* Inline user profile admin into user admin
* Add fields to user profile edit form
* Misc formatting changes from editor formatting (reversible if unwanted)
* Add user detail view + edit pages to show these collected values to users
* Also removes leftover .DS_Store files

## Discussion
* Should we ask existing users to update their profiles with the new fields?
* As @rgasper pointed out, the ZIP fields should be foreign keys to the ZIP code table. Likewise with states.

### Registration form
Screenshot 2023-05-19 at 4 13 36 PM

### Update profile form
Screenshot 2023-05-19 at 4 13 10 PM

### Admin
Screenshot 2023-05-19 at 4 14 34 PM

## User detail view
Screenshot 2023-05-26 at 9 37 17 AM

## User detail edit
Screenshot 2023-05-26 at 9 24 58 AM

### CODE REVIEW

File: caim_base/forms.py [hunk 1]
```diff
@@ -1,24 +1,55 @@
...
 from django.contrib.auth.forms import UserCreationForm
+from django.core.exceptions import ValidationError
+from django.db.models import Q
+
 from .models.awg import User
+from .models.geo import ZipCode
+
+
+def zip_validator(zip_code: str):
+    if not ZipCode.objects.filter(Q(zip_code=zip_code)).exists():
```
Comment (rgasper) [inline caim_base/forms.py:12]: This seems to me to be replicating what a Foreign Key to a ZipCode table would handle, right?

Is this simpler/better than the alternative for some reason I'm unaware of? Or are they pretty much equivalent for this situation.

File: caim_base/templates/user_profile/view.html [hunk 2]
```diff
@@ -1,37 +1,60 @@
...
-              &nbsp;
-            </p>
-          </div>
-        <div class="col-md-6 py-2 text-center text-md-end d-bottom-mobile">
-            {% if user == request.user %}
-                <a href="/user/{{ user.username }}/edit" class="btn btn-secondary">Edit profile</a>
-                <a href="{% url 'avatar_change' %}" class="btn btn-secondary">Change profile image</a>
-            {% endif %}
+      <div class="d-flex justify-content-center justify-content-sm-end">
+        <div class="text-center text-md-end d-flex flex-column flex-sm-row gap-2 py-2">
```
Comment (rgasper) [inline caim_base/templates/user_profile/view.html:7]: Ok so something that's been confusing me a little bit. This looks an awful lot like Tailwind CSS, but I don't see django-tailwind installed anywhere, and we're not running the tailwind "compiler" runserver command as part of the development processes. Could someone who knows a little more about CSS than I do explain what's going on here?

Comment (electric-al) [reply caim_base/templates/user_profile/view.html:7]: its boostrap5, which is heavily inspired by tailwind. https://getbootstrap.com/docs/5.0/getting-started/introduction/

No need for CSS compiling, as they are just CSS rules.

**Comment by electric-al** [comment]
Just a note. On the 'view public profile page' I dont think we should show real name or zip. this is for other public users. Real name / zip should only be visibile to admin or AWG admin staff.

File: caim_base/forms.py (hunk 1 above)
Comment (rwwiv) [reply caim_base/forms.py:12]: I was following the existing codebase, the Awg model uses a Charfield with a similar validator. 

Looking at moving both to a foreign key is worth doing at a later date for sure, but in this case I think consistency is key.

**Comment by rwwiv** [comment]
That's a very good point @electric-al. I didn't realize that was a public page. I'll just remove the values for now, but maybe we should look at toggling visibility if the user is viewing their own page or just creating a profile management page otherwise.

**Comment by rwwiv** [comment]
This is actually missing city and state and will require a little more work to be complete.

**Pull request merged**